app/(navigation)/(code)/components/Editor.tsx

```diff
@@ -137,37 +138,51 @@ function Editor() {
     textareaRef.current?.focus();
   });
 
-  useHotkeys("shift+option+f,shift+alt+f", (event) => {
+  useHotkeys("shift+option+f", (event) => {
     event.preventDefault();
     handleFormatCode();
   });
 
   const handleFormatCode = () => {
     const isSupportedLanguage = formatterSupportedLanguages.includes(selectedLanguage?.name || "");
     if (!isSupportedLanguage) {
-      return setFlashMessage({
-        message: "Formatting not supported for this language",
-        timeout: 1000,
-      });
+      return toast.error("Formatting is not supported for this language");
     }
     if (!code || !selectedLanguage) {
       return;
     }
     const language = selectedLanguage;
-    formatCode(code, language)
-      .then((formatted) => {
-        setCode(formatted);
-        // Sometimes hljs thinks the formatted code is a different language
-        // than the original, so we enforce the original language here
-        setSelectedLanguage(language);
-      })
-      .catch((e) => {
-        setFlashMessage({
-          message: "Formatting failed",
-          timeout: 1000,
-        });
-        return console.log("Formatting failed:", e);
-      });
+    toast.promise(
+      formatCode(code, language)
+        .then((formatted) => {
+          setCode(formatted);
+          // Sometimes hljs thinks the formatted code is a different language
+          // than the original, so we enforce the original language here
+          setSelectedLanguage(language);
+        })
+        .catch((e) => {
+          if (e.message.includes("Unexpected token")) {
+            toast.error("Formatting failed: Syntax Error", {
+              description: e.message,
+              duration: 3000,
+              dismissible: true,
+            });
+          } else {
+            toast.error("Code formatting failed", {
+              description: e.message,
+              duration: 3000,
+            });
+          }
+          return console.log("Formatting failed:", e);
+        }),
+      {
+        loading: "Formatting code...",
+        success: (data) => {
+          return "Successfully formatted code";
+        },
```

Comment:
saw that data wasn't used anyways so maybe?
```suggestion
        success: "Successfully formatted code",
```

Reply:
Thanks @harish-sethuraman, had this fixed in my local version!